adapter/repository/in-memory-simple/src/test/java/org/mobilitydata/gtfsvalidator/db/NoticeMemoryConsumptionTest.java

```diff
@@ -0,0 +1,163 @@
+package org.mobilitydata.gtfsvalidator.db;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mobilitydata.gtfsvalidator.usecase.notice.IntegerFieldValueOutOfRangeNotice;
+
+public class NoticeMemoryConsumptionTest {
+
+    private static final Logger LOGGER = LogManager.getLogger();
+    private static final int CONVERSION_FACTOR = 1_000_000;
+    private static final float BUFFER = 1.10f;
+
+    private void generateNotices(InMemoryValidationResultRepository resultRepository, int numberOfNotices) {
+        for (int i = 0; i < numberOfNotices; i++) {
+            resultRepository.addNotice(new IntegerFieldValueOutOfRangeNotice("filename",
+                    "fieldname",
+                    "entity_id",
+                    0,
+                    100,
+                    101)
+            );
+        }
+    }
+
+    @BeforeEach
+    public void cleanMemoryBeforeTest() {
+        System.gc();
+    }
+
+    @AfterEach
+    public void cleanMemoryAfterTest() {
+        System.gc();
+    }
+
+    @Test
+    public void creationOf100NoticeShouldNotExceedMemoryLimit() {
+
+        InMemoryValidationResultRepository resultRepository = new InMemoryValidationResultRepository();
+
+        generateNotices(resultRepository, 1);
+        resultRepository.getAll();
+
+        long totalMemory = Runtime.getRuntime().totalMemory(); // bytes
+        long freeMemory = Runtime.getRuntime().freeMemory();  // bytes
+
+        LOGGER.info(String.format("Generating 100 notices: Total memory: %s megabytes, Free memory: %s megabytes," +
```

Comment:
Should be a method with 3 parameters
`noticeCount`
`totalMemoryInBytes`
`freeMemoryInBytes`